webots_ros2_core/webots_ros2_core/utils.py

```diff
@@ -28,62 +28,69 @@
-    pass
 
+def get_required_webots_version():
+    """Return the Webots version compatible with this version of the package."""
+    return 'R2020b revision 1'
 
-def get_webots_home():
+
+def make_short_version(version):
+    """Transform a long Webots version in a short one."""
+    return version.replace('revision ', 'rev').replace(' ', '-')
+
+
+def get_required_webots_version_short():
+    """Return the Webots short version compatible with this version of the package."""
+    return make_short_version(get_required_webots_version())
+
+
+def get_webots_home(version=get_required_webots_version()):
     """Path to the Webots installation directory."""
-    webotsHome = None
-    if 'ROS2_WEBOTS_HOME' in os.environ:
-        webotsHome = os.environ['ROS2_WEBOTS_HOME']
-    elif ('webots_ros2_desktop' in sys.modules and
-          webots_ros2_desktop.webots_path and
-          webots_ros2_desktop.webots_path.get_webots_home()):
-        webotsHome = webots_ros2_desktop.webots_path.get_webots_home()
-    elif 'WEBOTS_HOME' in os.environ:
-        webotsHome = os.environ['WEBOTS_HOME']
-    elif os.path.isdir('/usr/local/webots'):  # Linux default install
-        webotsHome = '/usr/local/webots'
-    elif os.path.isdir('/snap/webots/current/usr/share/webots'):  # Linux snap install
-        webotsHome = '/snap/webots/current/usr/share/webots'
-    elif os.path.isdir('/Applications/Webots.app'):  # macOS default install
-        webotsHome = '/Applications/Webots.app'
-    elif os.path.isdir('C:\\Program Files\\Webots'):  # Windows default install
-        webotsHome = 'C:\\Program Files\\Webots'
-    elif os.path.isdir(os.getenv('LOCALAPPDATA') + '\\Programs\\Webots'):  # Windows user install
-        webotsHome = os.getenv('LOCALAPPDATA') + '\\Programs\\Webots'
-    else:
-        sys.exit('Webots not found, you should either define "ROS2_WEBOTS_HOME", "WEBOTS_HOME" ' +
-                 'or install the "webots_ros2_desktop" package.')
-    os.environ['WEBOTS_HOME'] = webotsHome
-    return webotsHome
+    environVariables = ['ROS2_WEBOTS_HOME', 'WEBOTS_HOME']
+    for variable in environVariables:
+        if variable in os.environ and os.path.isdir(variable) and get_webots_version(os.environ[variable]) == version:
+            os.environ['WEBOTS_HOME'] = os.environ[variable]
+            return os.environ[variable]
+    pathes = [
+        os.path.join(os.environ['HOME'], '.ros', 'webots' + make_short_version(version), 'webots'),
+        '/usr/local/webots',  # Linux default install
+        '/snap/webots/current/usr/share/webots',  # Linux snap install
+        '/Applications/Webots.app',  # macOS default install
+        'C:\\Program Files\\Webots',  # Windows default install
+        os.getenv('LOCALAPPDATA', '') + '\\Programs\\Webots'  # Windows user install
+    ]
+    for path in pathes:
+        if os.path.isdir(path) and get_webots_version(path) == version:
+            os.environ['WEBOTS_HOME'] = path
+            return path
```

Comment:
> In case Webots was installed in a different folder (from the tarball, or in D:\Program Files\Webots, etc.) this will fail.

In those cases, WEBOTS_HOME should already be set (manually from the user as stated in the user guide: https://www.cyberbotics.com/doc/guide/installation-procedure#installing-the-tarball-package or automatically from the installer on Windows).

> Instead of this, why don't we simply call "which webots" to determine which version is installed and going to be used?

Using which is a good idea, I would however not remove the other methods but add it, as the goal here is to find one specific version of Webots (the user may have several of them installed at different locations).